lib/payment/merchant_stripe.rb

```diff
@@ -0,0 +1,184 @@
+module ShorePayment
+  #
+  class StripeHash
+    def initialize(attrs = {})
+      update_attributes(attrs)
+    end
+
+    # Update object with this method. We have to take care updating 'nested'
+    #   objects
+    def update_attributes(attrs = {})
+      attrs.each_pair do |attr, value|
+        send(:"#{attr}=", value) if respond_to?(:"#{attr}=")
+      end if attrs
+    end
+  end
+
+  # Conversion between day of birth's Hash and Date representation
+  module StripeDobDate
+    def dob_date
+      if @dob && dob_present?
+        Date.new(@dob.year.to_i, @dob.month.to_i, @dob.day.to_i)
+      end
+    end
+
+    def dob_date=(new_date)
+      date = new_date.to_date if new_date.respond_to?(:to_date)
```

Comment:
I don't see the point of this ```if new_date.respond_to?(:to_date)```. And it will make the error you'd get on line 28 more obvious.